xmppserver/src/main/java/org/jivesoftware/database/SequenceManager.java

```diff
@@ -68,6 +68,7 @@
         new SequenceManager(JiveConstants.ROSTER, 5);
         new SequenceManager(JiveConstants.OFFLINE, 5);
         new SequenceManager(JiveConstants.MUC_ROOM, 5);
+        new SequenceManager(JiveConstants.MUC_MESSAGE_ID, 50);
```

Comment:
This essentially sets the maximum number of (unique) messages that can be stored in a batch to 50.

Reply:
50 is the number of id's to "checkout" at a time, so each openfire node reserves 50 future IDs... but if more are requested, the SequenceManager performs another query under the hood, to reserve more IDs, without duplicates. So, every time we ask for:
SequenceManager.nextID(JiveConstants.MUC_MESSAGE_ID)
we get a unique ID, even if the batch size is greater than 50.